GitUI/Avatars/FileSystemAvatarCache.cs

```diff
@@ -108,51 +106,40 @@ bool HasExpired()
                         return false;
                     }
 
-                    int cacheDays = AppSettings.AvatarImageCacheDays;
-                    if (cacheDays < 1)
+                    if (info.LastWriteTime < _expirationDate)
                     {
-                        cacheDays = DefaultCacheDays;
+                        TryDelete(path);
+                        return true;
                     }
 
-                    return info.LastWriteTime < DateTime.Now.AddDays(-cacheDays);
+                    return false;
                 }
+            }
+        }
 
-                void TryDelete()
-                {
-                    try
-                    {
-                        _fileSystem.File.Delete(path);
-                    }
-                    catch
-                    {
-                        // ignore
-                    }
-                }
+        private void TryDelete(string path)
+        {
+            try
+            {
+                _fileSystem.File.Delete(path);
+            }
+            catch
+            {
+                // ignore: if 2 instances are opened on the same repo, the 2nd one will prevent deletion, for example...
             }
         }
 
         /// <inheritdoc />
-        public async Task ClearCacheAsync()
+        public void ClearCache()
```

Comment:
❗ [perf] I/O operations must be performed async, so the API must remain async.

Reply:
I fully agree that `ClearCache` shall be run as background task.
But there is no need to deal with this in all implementations. Rather let the few callers do it right.

Reply:
I have removed the commit that remove the async/task because for me it was not the purpose of the PR (and at a beginning, that's you that ask for removing it and after reinstate it...)

Reply:
If asked for removing it when seeing the pointless async code.
But Igor's why, we force the callers to async calls. (I do not agree that `FileAndForget` was not clearly async. But it does not matter in this case.)